src/NuGet.Services.EndToEnd/Support/Clients/V2V3SearchClient.cs

```diff
@@ -0,0 +1,253 @@
+            }
+            else
+            {
+                searchBaseUrls.AddRange(await _v3IndexClient.GetSearchBaseUrls());
+            }
+
+            var v2SearchUrls = searchBaseUrls
+                .Select(u => $"{u}/search/query")
+                .ToList();
+
+            Assert.True(v2SearchUrls.Count > 0, "At least one search base URL must be configured.");
+
+            logger.WriteLine(
+                $"Waiting for package {id} {version} to be available on search endpoints:" + Environment.NewLine +
+                string.Join(Environment.NewLine, v2SearchUrls.Select(u => $" - {u}")));
+
+            var tasks = v2SearchUrls
+                .Select(u => WaitForPackageAsync(u, id, version, logger))
+                .ToList();
+
+            await Task.WhenAll(tasks);
+        }
+        
+        public async Task<IReadOnlyList<string>> GetSearchBaseUrlsAsync()
+        {
+            var searchBaseUrls = new List<string>();
+            if (_testSettings.SearchBaseUrl != null)
+            {
+                searchBaseUrls.Add(_testSettings.SearchBaseUrl);
+            }
+            else
+            {
+                searchBaseUrls.AddRange(await _v3IndexClient.GetSearchBaseUrls());
+            }
+
+            return searchBaseUrls;
+        }
+
+        private async Task WaitForPackageAsync(string v2SearchUrl, string id, string version, ITestOutputHelper logger)
+        {
+            await Task.Yield();
+
+            var url = QueryHelpers.AddQueryString(v2SearchUrl, "q", $"packageid:{id} and version:{version}");
+            url = QueryHelpers.AddQueryString(url, "ignoreFilter", "true");
+
+            var duration = Stopwatch.StartNew();
+            var found = false;
+            do
+            {
+                var response = await _httpClient.GetJsonAsync<V2SearchResponse>(url);
+                found = response
+                    .Data
+                    .Where(d => d.PackageRegistration?.Id == id && d.Version == version)
+                    .Any();
+
+                if (!found && duration.Elapsed + TestData.V3SleepDuration < TestData.V3WaitDuration)
+                {
+                    await Task.Delay(TestData.V3SleepDuration);
+                }
+            }
+            while (!found && duration.Elapsed < TestData.V3WaitDuration);
+
+            Assert.True(found, $"Package {id} {version} was not found on {v2SearchUrl} after waiting {TestData.V3WaitDuration}.");
+            logger.WriteLine($"Package {id} {version} was found on {v2SearchUrl} after waiting {duration.Elapsed}.");
+        }
+
+        public class V2SearchResponse
+        {
+            public List<V2SearchPackage> Data { get; set; }
+            public string Index { get; set; }
+            public long TotalHits { get; set; }
+        }
+
+        public class V2SearchPackage
+        {
+            public V2SearchPackageRegistration PackageRegistration { get; set; }
+            public string Version { get; set; }
```

Comment:
Please only add properties you are using and validating against. This is unnecessary boiler plate right now. (right?)